cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -138,9 +138,10 @@ def _add_resources(self):
         # Head Node EC2 Iam Role
         self._add_role_and_policies(self.config.head_node, "HeadNode")
 
-        # Compute Nodes EC2 Iam Roles
-        for queue in self.config.scheduling.queues:
-            self._add_role_and_policies(queue, queue.name)
+        if self._condition_is_slurm():
```

Comment:
we should move this logic in the SlurmConstruct if it's slurm specific

Reply:
I tried but it is not easy. To do so, we have to move `_add_role_and_policies` and related methods to `cdk_builder_utils`